builder/dockerfile/copy.go

```diff
@@ -82,14 +83,20 @@ func (o *copier) createCopyInstruction(args []string, cmdName string) (copyInstr
 	inst := copyInstruction{cmdName: cmdName}
 	last := len(args) - 1
 
-	// Work in daemon-specific filepath semantics
-	inst.dest = filepath.FromSlash(args[last])
+	// TODO: @gupta-ak. The platform needs to be here to support LCOW + WCOW. Or in the copier constructor.
+	separator := "/"
+	if !system.LCOWSupported() {
+		// Work in daemon-specific filepath semantics
+		inst.dest = filepath.FromSlash(args[last])
+		separator = string(os.PathSeparator)
+	}
```

Comment:
I added a platform tag to the `copier` struct. The platform was already in the `dispathRequest.builder`, so I just had to extract that into the`copierFromDistpatchRequest`. I added a `system.FromSlash()` and `system.Separator()` to check the directory suffix. 

If the copy instruction `dest` doesn't need the daemon/platform specific paths here, we could also just do a check like `if len(infos) > 1 && !strings.HasSuffix(filepath.ToSlash(inst.dest), "/")`